src/artemis/document.cljc

```diff
@@ -61,8 +61,8 @@
 (defn compose
   "Compose multiple parsed documents together to create a single document."
   {:added "0.1.0"}
-  [doc & docs]
-  (apply merge-with into doc docs))
+  [root-doc & docs]
```

Comment:
This was the crux of my witch hunt. I had composed multiple fragments, but we treat the first fragment as the main entry point, and my 'main fragment' was positioned last. So it seemed like deeply nested fragments were not working, when if you re-order the position the problem is solved. Changed the name of the first argument to make that mechanic more clear.